### ISSUE TITLE
[Bug]:  Stable-diffusionCan't run without a checkpoint.

### ISSUE THREAD

**Issue by dsthayl** [lshqqytiger/stable-diffusion-webui-directml#370]
### Checklist

- [ ] The issue exists after disabling all extensions
- [X] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [x] The issue exists in the current version of the webui
- [ ] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

No model is showing after i press refresh and no settings is seemingly saved, they all show error.
![thorium_g3sarp76aT_proc](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/159305724/57e83646-d922-4a74-901a-1cdd2a5a4a70)

### Steps to reproduce the problem

1. Open cmd to CD/location.
2. Git clone etc etc
3. Open web ui for the first time
4. Edit web ui file with no cuda arg
5. Error

### What should have happened?

Worked normally out of the box like was before.

### Sysinfo

Internal Server Error, even with --dump-sysinfo

### Console logs

```Shell
venv "C:\stabledf\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.7.0
Commit hash: adaea46e1c19d9a7091f89b0a7c6e66dfa732528
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\pytorch_lightning\utilities\distributed.py:258: LightningDeprecationWarning: `pytorch_lightning.utilities.distributed.rank_zero_only` has been deprecated in v1.8.1 and will be removed in v2.0.0. You can import it from `pytorch_lightning.utilities` instead.
  rank_zero_deprecation(
Launching Web UI with arguments: --skip-torch-cuda-test --opt-sub-quad-attention --medvram --disable-nan-check
Style database not found: C:\stabledf\stable-diffusion-webui-directml\styles.csv
Warning: caught exception 'Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx', memory monitor disabled
C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\diffusers\utils\outputs.py:63: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  torch.utils._pytree._register_pytree_node(
ONNX: selected=CUDAExecutionProvider, available=['AzureExecutionProvider', 'CPUExecutionProvider']
loading stable diffusion model: FileNotFoundError
Traceback (most recent call last):
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\initialize.py", line 147, in load_model
    shared.sd_model  # noqa: B018
  File "C:\stabledf\stable-diffusion-webui-directml\modules\shared_items.py", line 143, in sd_model
    return modules.sd_models.model_data.get_sd_model()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 537, in get_sd_model
    load_model()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 608, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 230, in select_checkpoint
    raise FileNotFoundError(error_message)
FileNotFoundError: No checkpoints found. When searching for checkpoints, looked at:
 - file C:\stabledf\stable-diffusion-webui-directml\model.ckpt
 - directory C:\stabledf\stable-diffusion-webui-directml\models\Stable-diffusionCan't run without a checkpoint. Find and place a .ckpt or .safetensors file into any of those locations.


Stable diffusion model failed to load
Applying attention optimization: sub-quadratic... done.
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 5.1s (prepare environment: 10.3s, initialize shared: 1.9s, load scripts: 1.0s, create ui: 0.8s, gradio launch: 0.5s).
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 152, in jsonable_encoder
    data = dict(obj)
TypeError: '_abc._abc_data' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 157, in jsonable_encoder
    data = vars(obj)
TypeError: vars() argument must have __dict__ attribute

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\applications.py", line 273, in __call__
    await super().__call__(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\gzip.py", line 24, in __call__
    await responder(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\gzip.py", line 44, in __call__
    await self.app(scope, receive, self.send_with_gzip)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\middleware\asyncexitstack.py", line 21, in __call__
    raise e
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\middleware\asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\routing.py", line 255, in app
    content = await serialize_response(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\routing.py", line 152, in serialize_response
    return jsonable_encoder(response_content)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 131, in jsonable_encoder
    jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\fastapi\encoders.py", line 160, in jsonable_encoder
    raise ValueError(errors) from e
ValueError: [TypeError("'_abc._abc_data' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
loading stable diffusion model: FileNotFoundError
Traceback (most recent call last):
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 419, in pages_html
    return refresh()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 425, in refresh
    pg.refresh()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks_textual_inversion.py", line 15, in refresh
    sd_hijack.model_hijack.embedding_db.load_textual_inversion_embeddings(force_reload=True)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 224, in load_textual_inversion_embeddings
    self.expected_shape = self.get_expected_shape()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 156, in get_expected_shape
    vec = shared.sd_model.cond_stage_model.encode_embedding_init_text(",", 1)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\shared_items.py", line 143, in sd_model
    return modules.sd_models.model_data.get_sd_model()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 537, in get_sd_model
    load_model()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 608, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 230, in select_checkpoint
    raise FileNotFoundError(error_message)
FileNotFoundError: No checkpoints found. When searching for checkpoints, looked at:
 - file C:\stabledf\stable-diffusion-webui-directml\model.ckpt
 - directory C:\stabledf\stable-diffusion-webui-directml\models\Stable-diffusionCan't run without a checkpoint. Find and place a .ckpt or .safetensors file into any of those locations.


Stable diffusion model failed to load
loading stable diffusion model: FileNotFoundError
Traceback (most recent call last):
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 419, in pages_html
    return refresh()
Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 425, in refresh
    pg.refresh()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks_textual_inversion.py", line 15, in refresh
    sd_hijack.model_hijack.embedding_db.load_textual_inversion_embeddings(force_reload=True)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 224, in load_textual_inversion_embeddings
    self.expected_shape = self.get_expected_shape()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 156, in get_expected_shape
    vec = shared.sd_model.cond_stage_model.encode_embedding_init_text(",", 1)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\shared_items.py", line 143, in sd_model
    return modules.sd_models.model_data.get_sd_model()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 537, in get_sd_model
    load_model()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 608, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 230, in select_checkpoint
    raise FileNotFoundError(error_message)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
FileNotFoundError: No checkpoints found. When searching for checkpoints, looked at:
 - file C:\stabledf\stable-diffusion-webui-directml\model.ckpt
 - directory C:\stabledf\stable-diffusion-webui-directml\models\Stable-diffusionCan't run without a checkpoint. Find and place a .ckpt or .safetensors file into any of those locations.

  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)

Stable diffusion model failed to load
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
loading stable diffusion model: FileNotFoundError
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 419, in pages_html
    return refresh()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 425, in refresh
    pg.refresh()
Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks_textual_inversion.py", line 15, in refresh
    sd_hijack.model_hijack.embedding_db.load_textual_inversion_embeddings(force_reload=True)
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 224, in load_textual_inversion_embeddings
    self.expected_shape = self.get_expected_shape()
  File "C:\Users\lucas\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 156, in get_expected_shape
    vec = shared.sd_model.cond_stage_model.encode_embedding_init_text(",", 1)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
AttributeError: 'NoneType' object has no attribute 'cond_stage_model'
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui.py", line 1816, in <lambda>
    visible=shared.sd_model and shared.sd_model.cond_stage_key == "edit"
  File "C:\stabledf\stable-diffusion-webui-directml\modules\shared_items.py", line 143, in sd_model
    return modules.sd_models.model_data.get_sd_model()
Traceback (most recent call last):
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 537, in get_sd_model
    load_model()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 608, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "C:\stabledf\stable-diffusion-webui-directml\modules\sd_models.py", line 230, in select_checkpoint
    raise FileNotFoundError(error_message)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
FileNotFoundError: No checkpoints found. When searching for checkpoints, looked at:
 - file C:\stabledf\stable-diffusion-webui-directml\model.ckpt
 - directory C:\stabledf\stable-diffusion-webui-directml\models\Stable-diffusionCan't run without a checkpoint. Find and place a .ckpt or .safetensors file into any of those locations.
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(

  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future

  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\stabledf\stable-diffusion-webui-directml\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 419, in pages_html
    return refresh()
Stable diffusion model failed to load
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks.py", line 425, in refresh
    pg.refresh()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\ui_extra_networks_textual_inversion.py", line 15, in refresh
    sd_hijack.model_hijack.embedding_db.load_textual_inversion_embeddings(force_reload=True)
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 224, in load_textual_inversion_embeddings
    self.expected_shape = self.get_expected_shape()
  File "C:\stabledf\stable-diffusion-webui-directml\modules\textual_inversion\textual_inversion.py", line 156, in get_expected_shape
    vec = shared.sd_model.cond_stage_model.encode_embedding_init_text(",", 1)
AttributeError: 'NoneType' object has no attribute 'cond_stage_model'
```

### Additional information

Im using the most recent AMD RX590 driver, no issues at all outside of this.

**Comment by lshqqytiger** [comment]
Do you have checkpoint files under `./models/Stable-diffusion`?

**Comment by Ziedelth** [comment]
Same error here with an AMD Radeon RX 6800. I have uninstalled venv folder and I have some models in Stable Diffusion folder

**Comment by charleskarol** [comment]
Same issue here. I just freshly re-installed SD (working well before) and is currently encountering the same problem.

**Comment by dsthayl** [comment]
> Do you have checkpoint files under `./models/Stable-diffusion`?

Yes, multiple .safetensor files

**Comment by Aria-NL** [comment]
Also encountering this issue, same log. UI is broken.

**Comment by VenomousValentine** [comment]
Same.

**Comment by trcyberoptic** [comment]
same for a completely new install

**Comment by VenomousValentine** [comment]
has this been fixed yet? I don't want to do a clean install until it is.

**Comment by lshqqytiger** [comment]
Could someone test again with latest commits? I can't reproduce this issue.

**Comment by dsthayl** [comment]
> Could someone test again with latest commits? I can't reproduce this issue.

Redid all steps, now in the second opening of webui-user (after --skip-cuda), a download for v1-5-pruned-emaonly.safetensors began (did not before).
I can now refresh and choose model and settings and use the program normally as always. I think its all fixed! Tested with multiple 1.5 (non XL) models and different settings and no problem showed up!
